frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -675,465 +919,376 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
+                />
+                {providedAPI.description}
               </div>
             )}
-          </Accordion>
-          {(!_.isEmpty(error) || !_.isEmpty(_.compact(_.values(formErrors)))) && (
             <Alert
               isInline
-              className="co-alert co-break-word co-alert--scrollable"
-              variant="danger"
-              title="Error"
-            >
-              {error || 'Fix above errors'}
-            </Alert>
-          )}
-          <div style={{ paddingBottom: '30px' }}>
-            <ActionGroup className="pf-c-form">
-              <Button onClick={submit} type="submit" variant="primary">
-                Create
-              </Button>
-              <Button onClick={history.goBack} variant="secondary">
-                Cancel
-              </Button>
-            </ActionGroup>
+              className="co-alert co-break-word"
+              variant="info"
+              title={
+                'Note: Some fields may not be represented in this form. Please select "Edit YAML" for full control of object creation.'
+              }
+            />
           </div>
-        </form>
-        <div className="col-md-4 col-lg-5">
-          {clusterServiceVersion && providedAPI && (
-            <div style={{ marginBottom: '30px' }}>
-              <ClusterServiceVersionLogo
-                displayName={providedAPI.displayName}
-                icon={_.get(clusterServiceVersion, 'spec.icon[0]')}
-                provider={_.get(clusterServiceVersion, 'spec.provider')}
-              />
-              {providedAPI.description}
-            </div>
-          )}
-          <Alert
-            isInline
-            className="co-alert co-break-word"
-            variant="info"
-            title={
-              'Note: Some fields may not be represented in this form. Please select "Edit YAML" for full control of object creation.'
-            }
-          />
         </div>
       </div>
-    </div>
+    </>
   );
 };
 
 /**
  * Component which wraps the YAML editor to ensure the templates are added from the `ClusterServiceVersion` annotations.
  */
-export const CreateOperandYAML: React.FC<CreateOperandYAMLProps> = (props) => {
-  const template = _.attempt(() => safeDump(props.sample));
+export const CreateOperandYAML: React.FC<CreateOperandYAMLProps> = ({
+  buffer,
+  clusterServiceVersion,
+  match,
+  operandModel,
+  onToggleEditMethod,
+}) => {
+  const template = React.useMemo(() => _.attempt(() => safeDump(buffer)), [buffer]);
   if (_.isError(template)) {
     // eslint-disable-next-line no-console
     console.error('Error parsing example JSON from annotation. Falling back to default.');
   }
+
+  const [parsedYAML, setParsedYAML] = React.useState(buffer);
+
+  const onYAMLChanged = (newYAML) => {
+    const newParsedYAML = _.attempt(() => safeLoad(newYAML));
+    setParsedYAML((currentParsedYAML) =>
+      !_.isError(newParsedYAML) ? newParsedYAML : currentParsedYAML,
+    );
+  };
```

Comment:
Some of this logic for parsing yaml into js should probably be in the CreateYaml component so that we can trigger it whenever we want instead of parsing it on every change. We should probably block  the user from toggling if the current yaml is invalid. Currently, we just attempt to parse the yaml on every change. If it's invalid yaml, then the buffer doesn't update. This could result in lost work if the user makes a lot of changes, but the yaml remains invalid the whole time.